x-pack/plugins/security_solution/public/common/components/alert_count_by_status/components/multiselect_popover/multiselect_popover.tsx

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback, useMemo, useState } from 'react';
+
+import { EuiFilterButton, EuiFilterGroup, EuiFilterSelectItem, EuiPopover } from '@elastic/eui';
+export interface MultiSelectPopoverProps {
+  title: string;
+  allItems: readonly string[];
+  selectedItems: string[];
+  onSelectedItemsChange: (newItems: string[]) => void;
+}
+
+export const MultiSelectPopover = React.memo(
+  ({ allItems, selectedItems, title, onSelectedItemsChange }: MultiSelectPopoverProps) => {
+    const [isItemPopoverOpen, setIsItemPopoverOpen] = useState(false);
+
+    const onChange = useCallback(
+      (item: string) => onSelectedItemsChange(getUpdatedSelectedItems(item, selectedItems)),
+      [selectedItems, onSelectedItemsChange]
+    );
+
+    const itemList = useMemo(() => {
+      return allItems.map((item, index) => (
+        <EuiFilterSelectItem
+          checked={selectedItems.includes(item) ? 'on' : undefined}
+          key={`${index}-${item}`}
+          onClick={() => onChange(item)}
+          title={item}
+        >
+          {item}
+        </EuiFilterSelectItem>
+      ));
+    }, [allItems, selectedItems, onChange]);
+
+    const togglePopover = useCallback(
+      (toState?: boolean) => {
+        setIsItemPopoverOpen((s) => (toState ? toState : !s));
+      },
+      [setIsItemPopoverOpen]
+    );
+
+    return (
+      <EuiPopover
+        ownFocus
+        button={
+          <EuiFilterGroup>
+            <EuiFilterButton
```

Comment:
@jamster10 As the docs show, EuiFilterGroup has to wrap the popover. Does the current code behave correctly?
![multi-select](https://user-images.githubusercontent.com/17747913/191480585-d889c781-a34f-4abc-976e-3e632750b775.png)

Reply:
Hey @semd! It appears to be working and looking as intended but I agree that it should follow Eui guideline. I have some tickets to add some more functionality and tweaks, and in there I will make this change. Thanks for pointing it out!